appjs/views/edit_project.js

```diff
@@ -20,12 +21,68 @@ function pluckAttr(models, attribute) {
 var EditProject = BaseView.extend(require('./mixins/actions'), require('./mixins/form'), {
   template: require('../templates/project.ejs'),
   events: {
-    'change :input': 'stopListeningForChanges'
+    'change :input': 'stopListeningForChanges',
+    'change form': 'debounceChange',
+    'keypress': 'debounceChange',
+    'click #savePreview': 'savePreview'
+  },
+
+  debounceChange: _.debounce(function(e){
+    if ( this.model.blueprint.hasPreviewType('live') ){
+      this.pollChange();
+    }
+  }, 2000),
+
+  pollChange: function(e){
+    logger.debug('pollchange');
+    var view = this,
+        $form = this.$('#projectForm'),
+        data = $form.alpaca('get').getValue(),
+        base_url = this.model.url();
+
+    if( view.model.isNew() ){
+      base_url = window.location.href;
+    }
+
+    var childLoaded = function() {
+      view.pollChange();
+    };
+
+    $.ajax({
+      type: "POST",
+      url: base_url + "/update_project_data",
+      data: JSON.stringify(data),
+      contentType: 'application/json',
+      dataType: 'json'
+    }).done(function( data ) {
+        logger.debug('received form values', data);
+
+        if(data.theme !== view.theme){
+          view.theme = data.theme;
+          var slug = view.model.blueprint.get('slug'),
+              bp_version = view.model.getVersion(),
+              preview_url = view.model.blueprint.getMediaUrl(
+                [bp_version, view.theme].join('-') + '/preview');
+
+          $('#'+slug+'__graphic').empty();
+          view.pym = new pym.Parent(slug+'__graphic', preview_url);
+          view.pym.iframe.onload = childLoaded;
+        }
+
+        view.pym.sendMessage('updateData', JSON.stringify(data));
+    });
+  },
+
+  savePreview: function(){
+    this.$('#projectForm form').submit();
   },
 
   afterInit: function(options) {
     this.disableForm = options.disableForm ? true : false;
     this.copyProject = options.copyProject ? true : false;
+    if(options.query){
+      this.togglePreview = options.query.togglePreview ? true : false;
```

Comment:
It's so that clicking the eye/preview icon for a project on the project list page opens the project's preview tab in Autotune, rather than opening a new browser tab for the preview.